test/api/attributes/[attributeId].test.ts

```diff
@@ -0,0 +1,171 @@
+import { ObjectId } from 'mongodb'
+import AttributeSchema, { AttributeDocument } from 'server/models/Attribute'
+import { ApiError, AttributeResponse } from 'utils/types'
+import { createRequest, createResponse } from 'node-mocks-http'
+import handler from 'pages/api/attributes/[attributeId]'
+import * as auth from 'utils/auth'
+import * as MongoDriver from 'server/actions/MongoDriver'
+import * as apiValidator from 'utils/apiValidators'
+import mongoose from 'mongoose'
+import { clientPromise } from '@api/auth/[...nextauth]'
+
+beforeAll(() => {
+  jest.spyOn(auth, 'serverAuth').mockImplementation(() => Promise.resolve())
+
+  jest.spyOn(apiValidator, 'apiObjectIdValidation').mockImplementation()
+})
+
+// restore mocked implementations and close db connections
+afterAll(() => {
+  jest.restoreAllMocks()
+  mongoose.connection.close()
+  clientPromise.then((client) => client.close())
+})
+
+describe('api/attributes/[attributeId]', () => {
+  test('thrown error is caught, response is unsuccessful and shows correct error message', async () => {
+    const mockObjectId = '6408a7156668c5655c25b105'
+
+    jest.spyOn(auth, 'serverAuth').mockImplementationOnce(async () => {
+      throw new ApiError(401, 'Unauthorized')
```

Comment:
Might make this a bit more maintainable if you assign the status code and error message to consts, so the passed in values and expected values aren't hard-coded. Just to ensure when you're writing the test that you don't make a typo. This can be done in all tests. Up to you if you want to do this.